### ISSUE TITLE
GCP - CVO HA  - code: 403, message: User is not allowed to do operation

### ISSUE THREAD

**Issue by tbeugin** [NetApp/terraform-provider-netapp-cloudmanager#198]
Hello,

We have an issue with our terraform code.

We don't use the connector installation in terraform due to the usage of service account from the git repo.
So we have deploy it from blue XP and the CVO HA on GCP trough our terraform code .

I have got an issue with the connector. We have redeploy an it from the bluxp console and Rediscover the CVO from the new connector.

Now we test the code updated with the new connector_id and we have this : 

```
Step #2 - "tf plan": Error: code: 403, message: User is not allowed to do operation
Step #2 - "tf plan":
Step #2 - "tf plan":   with netapp-cloudmanager_cvo_gcp.cvo-gcp-ha,
Step #2 - "tf plan":   on dev_netapp.tf line 52, in resource "netapp-cloudmanager_cvo_gcp" "cvo-gcp-ha":
Step #2 - "tf plan":   52: resource "netapp-cloudmanager_cvo_gcp" "cvo-gcp-ha" {
```

```
Step #2 - "tf plan": 2024-05-27T15:52:02.264Z [DEBUG] provider.terraform-provider-netapp-cloudmanager_v24.5.0: 2024/05/27 15:52:02 received: POST https://api.bluexp.netapp.com/auth/oauth/token 200 "{\"access_token\":\"XXXXXXXXX\",\"expires_in\":86400,\"token_type\":\"Bearer\"}"
Step #2 - "tf plan": 2024-05-27T15:52:02.264Z [DEBUG] provider.terraform-provider-netapp-cloudmanager_v24.5.0: 2024/05/27 15:52:02 Sending HTTP request: GET https://api.bluexp.netapp.com/occm/api/working-environments/exists/devcvoha map[string]interface {}{"body":map[string]interface {}(nil)}
Step #2 - "tf plan": 2024-05-27T15:52:02.368Z [DEBUG] provider.terraform-provider-netapp-cloudmanager_v24.5.0: 2024/05/27 15:52:02 received: GET https://api.bluexp.netapp.com/occm/api/working-environments/exists/devcvoha 403 "User is not allowed to do operation"
Step #2 - "tf plan": 2024-05-27T15:52:02.368Z [DEBUG] provider.terraform-provider-netapp-cloudmanager_v24.5.0: 2024/05/27 15:52:02 findWorkingEnvironmentByName request failed: User is not allowed to do operation
Step #2 - "tf plan": 2024-05-27T15:52:02.368Z [DEBUG] provider.terraform-provider-netapp-cloudmanager_v24.5.0: 2024/05/27 15:52:02 Error getting cvo
Step #2 - "tf plan": 2024-05-27T15:52:02.369Z [ERROR] vertex "netapp-cloudmanager_cvo_gcp.cvo-gcp-ha" error: code: 403, message: User is not allowed to do operation
Step #2 - "tf plan": 2024-05-27T15:52:02.369Z [ERROR] vertex "netapp-cloudmanager_cvo_gcp.cvo-gcp-ha (expand)" error: code: 403, message: User is not allowed to do operation
Step #2 - "tf plan": 2024-05-27T15:52:02.36
```

We don't have an issue for our 3 other CVO HA ( connector not réinstalled ) .

Best regards

Thomas

**Comment by tbeugin** [comment]
We have also test from Netapp service account instead of refresh token .

**Comment by tbeugin** [comment]
![image](https://github.com/NetApp/terraform-provider-netapp-cloudmanager/assets/155530544/ba909c66-eb02-4685-ae4a-94432f5c83b5)

Query is ok on postman with same id connector id as terraform code.

**Comment by tbeugin** [comment]
The only difference between the postman and the provider is the url to get the token 

On postman i use https://netapp-cloud-account.auth0.com/oauth/token instead of https://api.bluexp.netapp.com/auth/oauth/token (terraform)


 Sending HTTP request: POST https://api.bluexp.netapp.com/auth/oauth/token map[string]interface {}{"body":map[string]interface {}{"audience":"https://api.cloud.netapp.com", "client_id":"XXXX", "client_secret":"XXXXX", "grant_type":"client_credentials", "refresh_token":""}}

**Comment by suhasbshekar** [comment]
can you try https://api.bluexp.netapp.com/auth/oauth/token in postman? 
if it succeeds, use that token to GET the CVO?

**Comment by tbeugin** [comment]
Hello,

https://api.bluexp.netapp.com/auth/oauth/token  not use the same config as netapp-cloud-account for service account.
So i's not working with the change of url

Do you have the doc and the refresh token  for api.bluexp.netapp ? Or the postman config?

**Comment by tbeugin** [comment]
so done with the curl.

![image](https://github.com/NetApp/terraform-provider-netapp-cloudmanager/assets/155530544/c138e35a-c876-4a1f-a758-5e0717d7e7f3)

Token generated by curl also.
And request is ok.

So don't know where is the différence between the terraform and the curl request ...

**Comment by tbeugin** [comment]
Trouble solve 

I will edit this post for explanation

**Comment by tbeugin** [comment]
Hello,

The issue has finally been resolved.

Our query with Postman was successful with the new client IDs.

We checked the provider's code and noticed it was not using the netapp-cloud-account URL as mentioned in the NetApp documentation.

![image](https://github.com/NetApp/terraform-provider-netapp-cloudmanager/assets/155530544/0538d466-64d4-4a37-b96e-48f2e73f1048)





So, we recompiled the providers and tested with that, but the result was the same: a 403 error, "user denied."

The problem was much simpler.
Even though we changed the client_id in the Terraform code, Terraform was using the old client_id from the Terraform state to check the status of CVO.
Since the old client_id no longer exists, it returned a 403 error, "user denied."






So, we edited the Terraform state to change the client_id to the new one, and after that, Terraform worked correctly.
